### ISSUE TITLE
Hi, the model you implemented is not suited for cifar dataset?

### ISSUE THREAD

**Issue by eriche2016** [Mrliukuang/Torcifar#1]
It is the model which are used to training on imagenet dataset, which every image is of size 3x224x224, however, for cifar, its 3x32x32.  Have you run this net on cifar, if so, can you provide a copy of the net you used on cifar net?

**Comment by kuangliu** [comment]
They are modified version of vggnet, resnet and squeezenet for cifar10. 
Notice the output dimension is 10 (for cifar10) instead of 1000 (for imagenet). 
And yes, I've run them for like 50 times...

**Comment by eriche2016** [comment]
Hi, Kuangliu, see script [SqueezeNet.lua](https://github.com/kuangliu/torch-cifar/blob/master/SqueezeNet.lua#L41), it is designed to take input of size batchSize x 3 x 224x224, however,  for cifar-10, it is 3x32x32(https://github.com/kuangliu/torch-cifar/blob/master/provider.lua#L45). How you handle this case?

**Comment by kuangliu** [comment]
Yeah, you're right. 
The squeezenet input is 224x224. If you are interested in a cifar10 version of squeezenet, try to remove the AvgPool layer, should be working.

**Comment by eriche2016** [comment]
actually, i am thinking about resize each image in cifar10(preprocessing) to the size of 3x224x224, or just do it by inserting a `resize` layer to the model and resize it at runtime. It should be working.  And removing avgPool is also a good  choice here. I will see it. thank you very much for you quick reply.

**Comment by eriche2016** [comment]
Hi, kuangliu, i trained the model of squeezeNet, but the can hardly see any improvements on the training error as time goes on, what is the matter? what about your experiments on cifar-10?

**Comment by kuangliu** [comment]
Did you perform normalization in provider.lua?
Make sure you uncomment [line 48](https://github.com/kuangliu/torch-cifar/blob/master/provider.lua#L48) in provider.lua. I comment this out for debug reasons, cause preprocessing is really slow.

Update:
I just tried, and managed to get squeezenet working fine with [MSRinit](https://github.com/kuangliu/xtorch/blob/datagen/utils.lua#L142).
And set `learningRate` to `0.001`.

**Comment by kuangliu** [comment]
I manage to get squeezenet working fine with [MSRinit](https://github.com/kuangliu/xtorch/blob/datagen/utils.lua#L142).
And set `learningRate` to `0.001`.

**Comment by eriche2016** [comment]
yes, i am not sure whether you use squeezeNet to training your model. And i also not use normalize() in file `provider.lua`. But it doesnot work. Also i try to use MSRinit following your advice. The answer is also negative, and cannot converge.  Can you give more tips? what is your solution?

update: 
Also `uncomment` normalize in `provider.lua` is kinda confusing, 'cause for residual network, i am not use 
it. Even if i follow your  advice of `uncomment` the line in `provider.lua` , also cannot converge.

**Comment by kuangliu** [comment]
The following steps works for me:
1. **Uncomment** [line 48](https://github.com/kuangliu/torch-cifar/blob/master/provider.lua#L48) in `provider.lua`. Data normalization is important, make sure it works.
2. Add MSRinit to squeezenet.
3. Decrease learning rate to 0.001.

**Comment by eriche2016** [comment]
thanks for your reply. But I use normalization steps like [cifar.torch](https://github.com/szagoruyko/cifar.torch).  Also i add MSRint to squeezenet. Thirdly, i decrease learning rate to 0.001

**Comment by eriche2016** [comment]
Also, note that i just use your [model script](https://github.com/kuangliu/torch-cifar/blob/e7cccbb3509f1af8b846f66afcdd25348319b633/SqueezeNet.lua).

**Comment by eriche2016** [comment]
Hi,  kuangliu, i use your code and run experiments on cifar, still i donot see any improvements on the training accuracy, i,e. cannnot converge.  Have you noticed the same phenomina?

**Comment by kuangliu** [comment]
I just upload the code&data to: http://pan.baidu.com/s/1qXDTFwK
download it and run `th main_multiGPU.lua`
sometimes you may get `loss = nan`, just run it again.

**Comment by eriche2016** [comment]
thank you very much for providing the code and data. I do get `loss = non` some time. How to avoid such a problem.   Besides, the performance of squeezeNet  is not that competitive. 
By the way, have you tried using `train.lua` to do experiments on squeezenet, to me, it cannot converge using script `train.lua`
